packages/relay/src/lib/eth.ts

```diff
@@ -1578,46 +1583,24 @@ export class EthImpl implements Eth {
         );
 
       txSubmitted = true;
-      fileId = sendRawTransactionResult!.fileId;
-
-      // Wait for the record from the execution.
-      const txId = sendRawTransactionResult!.txResponse.transactionId.toString();
-      const formattedId = formatTransactionIdWithoutQueryParams(txId);
-
-      // handle formattedId being null
-      if (!formattedId) {
-        throw predefined.INTERNAL_ERROR(`Invalid transactionID: ${txId}`);
-      }
-
-      const contractResult = await this.mirrorNodeClient.repeatedRequest(
-        this.mirrorNodeClient.getContractResult.name,
-        [formattedId, requestDetails],
-        this.mirrorNodeClient.getMirrorNodeRequestRetryCount(),
-        requestDetails,
-      );
-
-      if (!contractResult) {
-        this.logger.warn(`${requestIdPrefix} No record retrieved`);
-        throw predefined.INTERNAL_ERROR(`No matching record found for transaction id ${txId}`);
-      }
-
-      if (contractResult.hash == null) {
-        this.logger.error(
-          `${requestIdPrefix} The ethereumHash can never be null for an ethereum transaction, and yet it was!!`,
+      fileId = sendRawTransactionResult.fileId;
+      submittedTransactionId = sendRawTransactionResult.txResponse.transactionId?.toString();
+      formattedTransactionId = formatTransactionIdWithoutQueryParams(submittedTransactionId);
+      if (!formattedTransactionId) {
+        this.logger.warn(
+          `${requestIdPrefix} Transaction successfully submitted but returned invalid transactionID: transactionId==${submittedTransactionId}`,
+        );
+        throw predefined.INTERNAL_ERROR(
+          `Transaction successfully submitted but returned invalid transactionID: transactionId==${submittedTransactionId}`,
```

Comment:
Also here, why are we logging a warning here if we are immediately throwing an error with exactly the same message?

Reply:
https://github.com/hashgraph/hedera-json-rpc-relay/pull/3229#discussion_r1834622780

Reply:
https://github.com/hashgraph/hedera-json-rpc-relay/pull/3229#discussion_r1836126524

Reply:
Updated in https://github.com/hashgraph/hedera-json-rpc-relay/pull/3229/commits/9f9f1622d8b4e3c692abe19ae86e5ca80399e08e. Thanks for the suggestion!